bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/protocol/servers/BindingServlet.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linktap.protocol.servers;
+
+import static org.openhab.binding.linktap.protocol.frames.TLGatewayFrame.EMPTY_STRING;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.core.MediaType;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.linktap.internal.LinkTapBindingConstants;
+import org.openhab.binding.linktap.internal.TransactionProcessor;
+import org.openhab.binding.linktap.protocol.frames.TLGatewayFrame;
+import org.osgi.service.http.HttpService;
+import org.osgi.service.http.NamespaceException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BindingServlet} defines the request to enable or disable alerts from a given device.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class BindingServlet extends HttpServlet {
+    private static final long serialVersionUID = -23L;
+
+    private final Logger logger = LoggerFactory.getLogger(BindingServlet.class);
+
+    public static final String SERVLET_URL_WITHOUT_ROOT = "linkTap";
+    private static final String SERVLET_URL = "/" + SERVLET_URL_WITHOUT_ROOT;
+    @Nullable
+    HttpService httpService;
+
+    volatile boolean registered;
+    final Object registerLock = new Object();
+
+    public static final BindingServlet INSTANCE = new BindingServlet();
+
+    public static final BindingServlet getInstance() {
+        return INSTANCE;
+    }
+
+    public void setHttpService(final HttpService httpService) {
+        this.httpService = httpService;
+    }
+
+    public static String getServletAddress(final String hostname) {
+        final String httpPortStr = System.getProperty("org.osgi.service.http.port");
+        final String httpsPortStr = System.getProperty("org.osgi.service.https.port");
+        final Logger logger = LoggerFactory.getLogger(BindingServlet.class);
+        if (httpPortStr == null || httpPortStr.isEmpty()) {
+            logger.warn("HTTP Server port is not running, cannot use for API callbacks");
+            if (httpsPortStr != null && !httpsPortStr.isEmpty()) {
+                logger.warn("It looks like HTTPS is enabled only - the device needs HTTP for efficient comm's");
```

Comment:
It's purely because of the resources on the gateways embedded chip as I did query if I could use https as it wasn't in any of the protocols tech specs. So at the moment the gateway only supports http. The warning was there so people knew the system has had to fallback basically to polling only. (I wrap my OH http port up via Nginx but allow the firewall only to allow the particular device, to http for this URL). I would hope most people are running in a protected environment something likely critical such as OH :) That was the logic why, and why https isn't the defacto standard.

Reply:
I would opt to remove the inner check for http. The outer check + log about http should already be enough to warn the user.

Reply:
I have removed to only the outer one simplification is always good.